File: src/test/java/cz/jskrabal/proxy/config/enums/ConfigurationParameterTest.java
package cz.jskrabal.proxy.config.enums;

import java.util.List;

import org.junit.Test;
import static org.junit.Assert.assertEquals;

/**
 * Created by janskrabal on 15/06/16.
 */
public class ConfigurationParameterTest {

	@Test
	public void testGetFullJsonKey() {
		String[] fullKeyParts;
		List<String> keyParts;

		for (ConfigurationParameter parameter : ConfigurationParameter.values()) {
			keyParts = parameter.getJsonKeyParts();
            fullKeyParts = parameter.getFullJsonKey().split("\\.");

            assertEquals(keyParts.size(), fullKeyParts.length);

			for (int i = 0; i < keyParts.size(); i++) {
                assertEquals(keyParts.get(i), fullKeyParts[i]);
			}
		}
	}

}

File: src/test/kotlin/cz/jskrabal/proxy/config/enums/ConfigurationParameterTest.kt
package cz.jskrabal.proxy.config.enums

import org.junit.Assert.assertEquals

import org.junit.Test

/**
 * Created by janskrabal on 15/06/16.
 */
class ConfigurationParameterTest {

    @Test
    fun testGetFullJsonKey() {
        var fullKeyParts: Array<String>
        var keyParts: List<String>

        for (parameter in ConfigurationParameter.values()) {
            keyParts = parameter.jsonKeyParts
            fullKeyParts = parameter.fullJsonKey.split("\\.".toRegex()).dropLastWhile { it.isEmpty() }.toTypedArray()

            assertEquals(keyParts.size.toLong(), fullKeyParts.size.toLong())

            for (i in keyParts.indices) {
                assertEquals(keyParts[i], fullKeyParts[i])
            }
        }
    }

}
